### PR TITLE
[Do not merge] Test PR used for pair-review

### PR DESCRIPTION
**Pull request by zhengchang907**
This is a test PR for pair-review

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -4,6 +4,7 @@ on:
     branches:
       - develop
       - 20201202-hotfix
+      - zhengchang-cert-only
```
Comment (zhengchang907) [inline .github/workflows/build.yml]: delete
 - 20201202-hotfix
 - zhengchang-cert-only

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -25,7 +26,7 @@ env:
   nsg: wls-nsg
   numberOfInstances: 2
   refArmttk: 8e08a92dbb0a22b8560ec854a60e9a8a7b71c68f
-  refJavaee: 6807e2ae3e8a8492edf211b9f4f0b7ed3a2a29e7
+  refJavaee: update-for-weblogic-cluster
```
Comment (zhengchang907) [inline .github/workflows/build.yml:27]: replace with the official latest commit hashcode

File: arm-oraclelinux-wls-cluster/pom.xml [hunk 3]
```diff
@@ -41,8 +41,8 @@
     <properties>
         <template.validation.tests.directory>${basedir}/../../arm-ttk/arm-ttk</template.validation.tests.directory>
         <test.parameters>-TestParameter '@{&quot;PasswordMinLength&quot;=6}'</test.parameters>
-        <template.pid.properties.url>https://raw.githubusercontent.com/wls-eng/arm-oraclelinux-wls/develop/src/main/resources/pid.properties</template.pid.properties.url>
-        <template.microsoft.pid.properties.url>https://raw.githubusercontent.com/wls-eng/arm-oraclelinux-wls/develop/src/main/resources/microsoft-pid.properties</template.microsoft.pid.properties.url>
+        <template.pid.properties.url>https://raw.githubusercontent.com/galiacheng/arm-oraclelinux-wls/develop/src/main/resources/pid.properties</template.pid.properties.url>
```
Comment (zhengchang907) [inline arm-oraclelinux-wls-cluster/pom.xml:44]: restore

File: arm-oraclelinux-wls-cluster/src/main/arm/createUiDefinition.json [hunk 4]
```diff
@@ -277,13 +277,72 @@
                         "type": "Microsoft.Common.TextBlock",
                         "visible": "[steps('section_appGateway').enableAppGateway]",
                         "options": {
-                            "text": "App Gateway integration requires an SSL Certificate to enable SSL termination at the gateway.  End-to-end SSL encryption is not supported by the template.  The template will look for the certificate and its password in the Azure KeyVault specified here.",
+                            "text": "App Gateway integration requires a SSL certificate to enable SSL termination at the gateway.  There are three SSL certificate usage available. Upload a SSL certificate: the template will store the certificate to an Azure Key Vault and retrieve it during the gateway deployment.  Have a Azure Key Vault with a SSL certificate stored: the template will look for the certificate and its password in the Azure KeyVault specified here. Generate a self-signed certificate: the tamplate will generate a self signed certificate and apply it during the gateway deployment.",
```
Comment (zhengchang907) [inline arm-oraclelinux-wls-cluster/src/main/arm/createUiDefinition.json]: 'There are three SSL certificate usage[s]'
'Key[space]Vault'
'ta[e]mplate'
'self[-]signed'
'have a[n] Azure Key Vault'

File: arm-oraclelinux-wls-cluster/src/main/arm/createUiDefinition.json [hunk 5]
```diff
@@ -277,13 +277,72 @@
...
+                        "defaultValue": "Upload a SSL certificate",
+                        "toolTip": "Select your current usage of SSL Certificate",
+                        "constraints": {
+                            "allowedValues": [
+                                {
+                                    "label": "Upload a SSL certificate",
+                                    "value": "haveCert"
+                                },
+                                {
+                                    "label": "Have a Azure Key Vault with a SSL certificate stored",
```
Comment (zhengchang907) [inline arm-oraclelinux-wls-cluster/src/main/arm/createUiDefinition.json]: 'Have a[n] Azure Key Vault'

File: arm-oraclelinux-wls-cluster/src/main/arm/createUiDefinition.json [hunk 6]
```diff
@@ -277,13 +277,72 @@
...
+                            ],
+                            "required": true
+                        },
+                        "visible": "[steps('section_appGateway').enableAppGateway]"
+                    },
+                    {
+                        "name": "keyVaultSSLCertData",
+                        "type": "Microsoft.Common.FileUpload",
+                        "label": "SSL certificate",
+                        "toolTip": "SSL certificate used for App Gateway",
```
Comment (zhengchang907) [inline arm-oraclelinux-wls-cluster/src/main/arm/createUiDefinition.json:316]: "SSL certificate [(.pfx)]",

File: arm-oraclelinux-wls-cluster/src/main/arm/createUiDefinition.json [hunk 7]
```diff
@@ -334,11 +393,238 @@
...
-                        "visible": "[steps('section_appGateway').enableAppGateway]"
+                        "visible": "[equals(steps('section_appGateway').certificateOption, 'haveKeyVault')]"
+                    },
+                    {
+                        "name": "infoGatewayIndentity",
+                        "type": "Microsoft.Common.InfoBox",
+                        "visible": "[and(equals(steps('section_appGateway').certificateOption, 'generateCert'), less(length(steps('section_appGateway').gatewayIdentity.userAssignedIdentities), 1))]",
+                        "options": {
+                            "icon": "Error",
+                            "text": "This option will create a self signed SSL certificate for gateway SSL termination. We use Azure deployment scripts to create a certificate. You have to add at least one user assigned identity to access Azure resources.",
```
Comment (zhengchang907) [inline arm-oraclelinux-wls-cluster/src/main/arm/createUiDefinition.json]: self[-]signed

File: arm-oraclelinux-wls-cluster/src/main/arm/createUiDefinition.json [hunk 8]
```diff
@@ -334,11 +393,238 @@
...
+                        "type": "Microsoft.Common.Section",
+                        "label": "DNS Configuration Settings",
+                        "visible": "[bool(steps('section_dnsConfiguration').enableCustomDNS)]",
+                        "elements": [
+                            {
+                                "name": "bringDNSZone",
+                                "type": "Microsoft.Common.OptionsGroup",
+                                "label": "Bring your own Azure DNS Zone",
+                                "defaultValue": "No",
+                                "toolTip": "Select 'Yes' to configure Custom DNS Alias based on your existing Azure DNS Zone, 'No' to create a Azure DNS Zone.",
```
Comment (zhengchang907) [inline arm-oraclelinux-wls-cluster/src/main/arm/createUiDefinition.json]: a[n] Azure DNS Zone

File: arm-oraclelinux-wls-cluster/src/main/arm/createUiDefinition.json [hunk 9]
```diff
@@ -334,11 +393,238 @@
...
+                                        "label": "Learn more",
+                                        "uri": "https://aka.ms/dns-domain-delegation"
+                                    }
+                                }
+                            },
+                            {
+                                "name": "deplymentScriptInfo",
+                                "type": "Microsoft.Common.TextBlock",
+                                "options": {
+                                    "text": "Bring Azure DNS Zone option will add records to your DNS Zone to cause Application Gateway accessible. We need a user-assigned managed identity to update your resource group that has Azure DNS Zone deployed. ",
```
Comment (zhengchang907) [inline arm-oraclelinux-wls-cluster/src/main/arm/createUiDefinition.json]: to cause Application Gateway accessible [via DNS alias]

File: arm-oraclelinux-wls-cluster/src/main/arm/createUiDefinition.json [hunk 10]
```diff
@@ -334,11 +393,238 @@
...
+                                "constraints": {
+                                    "required": true,
+                                    "regex": "^([0-9a-zA-Z_-]{1,63}\\.){1,33}[0-9a-zA-Z_-]{1,63}$",
+                                    "validationMessage": "There must be between 2 and 34 labels. For example, \"contoso.com\" has 2 labels. Each label must contain between 1 and 63 characters. Each label must only contain letters, numbers, underscores, and dashes."
+                                }
+                            },
+                            {
+                                "name": "dnsZoneResourceGroup",
+                                "type": "Microsoft.Common.TextBox",
+                                "label": "Resource group name in current subscription containing the DNS Zone",
```
Comment (zhengchang907) [inline arm-oraclelinux-wls-cluster/src/main/arm/createUiDefinition.json]: Name of the resource group contains the DNS Zone in current subscription

File: arm-oraclelinux-wls-cluster/src/main/arm/createUiDefinition.json [hunk 11]
```diff
@@ -334,11 +393,238 @@
...
+                                "defaultValue": "",
+                                "toolTip": "Resource group name in current subscription containing the DNS Zone",
```
Comment (zhengchang907) [inline arm-oraclelinux-wls-cluster/src/main/arm/createUiDefinition.json]: as above

File: arm-oraclelinux-wls-cluster/src/main/arm/createUiDefinition.json [hunk 12]
```diff
@@ -334,11 +393,238 @@
...
+                                            "isValid": "[less(sub(length(concat(steps('section_dnsConfiguration').customDNSSettings.dnszoneAdminConsoleLabel,'.',steps('section_dnsConfiguration').customDNSSettings.dnszoneName)),length(replace(concat(steps('section_dnsConfiguration').customDNSSettings.dnszoneAdminConsoleLabel,'.',steps('section_dnsConfiguration').customDNSSettings.dnszoneName), '.', ''))),34)]",
+                                            "message": "Subdomain must be between 2 and 34 labels. For example, \"applications.contoso.com\" has 3 labels."
+                                        }
+                                    ]
+                                }
+                            },
+                            {
+                                "name": "dnszoneGatewayLabel",
+                                "type": "Microsoft.Common.TextBox",
+                                "label": "Label for Oracle WebLogic Administration Console",
```
Comment (zhengchang907) [inline arm-oraclelinux-wls-cluster/src/main/arm/createUiDefinition.json]: Label for Application Gateway

File: arm-oraclelinux-wls-cluster/src/main/arm/createUiDefinition.json [hunk 13]
```diff
@@ -931,4 +1228,4 @@
             "wlsUserName": "[basics('basicsRequired').wlsUserName]"
         }
     }
-}
+}
```
Comment (zhengchang907) [inline arm-oraclelinux-wls-cluster/src/main/arm/createUiDefinition.json]: add new line

File: arm-oraclelinux-wls-cluster/src/main/arm/mainTemplate.json [hunk 14]
```diff
@@ -54,6 +54,41 @@
...
+                "userAssignedIdentities": {
+                    "/subscriptions/subscriptionId/resourceGroups/TestResourceGroup/providers/Microsoft.ManagedIdentity/userAssignedIdentities/TestUserIdentity1": {}
+                }
+            }
+        },
+        "appGatewaySSLCertData": {
+            "defaultValue": "",
+            "type": "string",
+            "metadata": {
+                "description": "The value of the SSL Certificate Data"
```
Comment (zhengchang907) [inline arm-oraclelinux-wls-cluster/src/main/arm/mainTemplate.json]: The one-line, base64 string of the SSL certificate data.

File: arm-oraclelinux-wls-cluster/src/main/arm/mainTemplate.json [hunk 15]
```diff
@@ -166,6 +245,10 @@
                 "description": "Public IP Name for the Application Gateway"
             }
         },
+        "hasDNSZones": {
+            "type": "bool",
```
Comment (zhengchang907) [inline arm-oraclelinux-wls-cluster/src/main/arm/mainTemplate.json:249]: add metadata

File: arm-oraclelinux-wls-cluster/src/main/arm/nestedtemplates/_dnszones/_createDNSZonesTemplate.json [hunk 16]
```diff
@@ -0,0 +1,116 @@
...
+        "dnszonesCNAMEAlias": {
+            "type": "array",
+            "metadata": {
+                "description": "An array of CNAME record alias. Used to create CNAME records."
+            }
+        },
+        "dnszonesCNAMERecordSetNames": {
+            "type": "array",
+            "metadata": {
+                "description": "An array of A record names. Used to create CNAME record set."
```
Comment (zhengchang907) [inline arm-oraclelinux-wls-cluster/src/main/arm/nestedtemplates/_dnszones/_createDNSZonesTemplate.json]: "An array of A[CNAME] record names. Used to create CNAME record set."

File: arm-oraclelinux-wls-cluster/src/main/arm/nestedtemplates/_dnszones/_createDNSZonesTemplate.json [hunk 17]
```diff
@@ -0,0 +1,116 @@
...
+            }
+        }
+    ],
+    "outputs": {
+        "dnszonesName": {
+            "type": "string",
+            "value": "[parameters('dnszoneName')]"
+        }
+    }
+}
```
Comment (zhengchang907) [inline arm-oraclelinux-wls-cluster/src/main/arm/nestedtemplates/_dnszones/_createDNSZonesTemplate.json]: new line

File: arm-oraclelinux-wls-cluster/src/main/arm/nestedtemplates/_dnszones/_updateDNSZonesTemplate.json [hunk 18]
```diff
@@ -0,0 +1,129 @@
...
+        "dnszonesCNAMEAlias": {
+            "type": "array",
+            "metadata": {
+                "description": "An array of CNAME record alias. Used to create CNAME records."
+            }
+        },
+        "dnszonesCNAMERecordSetNames": {
+            "type": "array",
+            "metadata": {
+                "description": "An array of A record names. Used to create CNAME record set."
```
Comment (zhengchang907) [inline arm-oraclelinux-wls-cluster/src/main/arm/nestedtemplates/_dnszones/_updateDNSZonesTemplate.json]: "description": "An array of A[CNAME] record names. Used to create CNAME record set."

File: arm-oraclelinux-wls-cluster/src/main/arm/nestedtemplates/_dnszones/_updateDNSZonesTemplate.json [hunk 19]
```diff
@@ -0,0 +1,129 @@
...
+        "location": {
+            "type": "string",
+            "metadata": {
+                "description": "Location for all resources."
+            }
+        },
+        "resourceGroup": {
+            "type": "string",
+            "metadata": {
+                "description": "The resource group name that has an Azure DNS Zone deployed."
```
Comment (zhengchang907) [inline arm-oraclelinux-wls-cluster/src/main/arm/nestedtemplates/_dnszones/_updateDNSZonesTemplate.json]: Name of the resource group that has an Azure DNS Zone deployed.

File: arm-oraclelinux-wls-cluster/src/main/arm/nestedtemplates/dnszonesTemplate.json [hunk 20]
```diff
@@ -0,0 +1,225 @@
...
+            "defaultValue": "[resourceGroup()]",
+            "metadata": {
+                "description": "The resource group name that has an Azure DNS Zone deployed."
+            }
+        },
+        "targetResources": {
+            "type": "array",
+            "defaultValue": [],
+            "metadata": {
+                "description": "References to azure resources from where the dns resource value is taken. Each item is corresponding to values of dnszonesARecordSetNames."
```
Comment (zhengchang907) [inline arm-oraclelinux-wls-cluster/src/main/arm/nestedtemplates/dnszonesTemplate.json]: Azure DNS

File: arm-oraclelinux-wls-cluster/src/main/arm/nestedtemplates/_dnszones/_createDNSZonesTemplate.json [hunk 21]
```diff
@@ -0,0 +1,116 @@
...
+        "location": {
+            "type": "string",
+            "metadata": {
+                "description": "Location for all resources."
+            }
+        },
+        "targetResources": {
+            "type": "array",
+            "metadata": {
+                "description": "References to azure resources from where the dns resource value is taken. Each item is corresponding to values of dnszonesARecordSetNames."
```
Comment (zhengchang907) [inline arm-oraclelinux-wls-cluster/src/main/arm/nestedtemplates/_dnszones/_createDNSZonesTemplate.json]: Azure DNS

File: arm-oraclelinux-wls-cluster/src/main/arm/nestedtemplates/_dnszones/_updateDNSZonesTemplate.json [hunk 22]
```diff
@@ -0,0 +1,129 @@
...
+            }
+        },
+        "targetResources": {
+            "type": "array",
+            "metadata": {
+                "description": "References to azure resources from where the dns resource value is taken. Each item is corresponding to values of dnszonesARecordSetNames."
```
Comment (zhengchang907) [inline arm-oraclelinux-wls-cluster/src/main/arm/nestedtemplates/_dnszones/_updateDNSZonesTemplate.json]: Azure DNS

File: arm-oraclelinux-wls-cluster/src/main/arm/nestedtemplates/_dnszones/_updateDNSZonesTemplate.json [hunk 23]
```diff
@@ -0,0 +1,129 @@
...
+                "AzCliVersion": "2.15.0",
+                "timeout": "PT30M",
+                "arguments": "[format('{0} {1} {2} {3} {4} {5} {6} {7} {8} {9} {10}', parameters('resourceGroup'), parameters('dnszoneName'), array.join(parameters('dnszonesARecordSetNames')),array.join(parameters('targetResources')), length(parameters('dnszonesARecordSetNames')), length(parameters('targetResources')), parameters('ttl'), array.join(parameters('dnszonesCNAMERecordSetNames')),array.join(parameters('dnszonesCNAMEAlias')),length(parameters('dnszonesCNAMERecordSetNames')), length(parameters('dnszonesCNAMEAlias')))]",
+                "primaryScriptUri": "[uri(parameters('_artifactsLocationDNSZonesTemplate'), concat('../../scripts/', variables('name_scriptDNSConfiguration'), parameters('_artifactsLocationSasToken')))]",
+                "cleanupPreference": "OnSuccess",
+                "retentionInterval": "P1D"
+            }
+        }
+    ]
+}
```
Comment (zhengchang907) [inline arm-oraclelinux-wls-cluster/src/main/arm/nestedtemplates/_dnszones/_updateDNSZonesTemplate.json]: new line

File: arm-oraclelinux-wls-cluster/src/main/arm/nestedtemplates/_keyvault/_keyvaultWithNewCertTemplate.json [hunk 24]
```diff
@@ -0,0 +1,189 @@
...
+            "location": "[parameters('location')]",
+            "properties": {
+                "sku": {
+                    "family": "A",
+                    "name": "[parameters('sku')]"
+                },
+                "tenantId": "[subscription().tenantId]",
+                "accessPolicies": [
+                    {
+                        "objectId": "[reference(string(identity.getId(parameters('identity'))), '2018-11-30').principalId]",
```
Comment (zhengchang907) [inline arm-oraclelinux-wls-cluster/src/main/arm/nestedtemplates/_keyvault/_keyvaultWithNewCertTemplate.json]: api version

File: arm-oraclelinux-wls-cluster/src/main/arm/nestedtemplates/_keyvault/_keyvaultWithNewCertTemplate.json [hunk 25]
```diff
@@ -0,0 +1,189 @@
...
+        "keyvaultName": {
+            "type": "string",
+            "value": "[parameters('keyvaultName')]"
+        },
+        "secretName": {
+            "type": "string",
+            "value": "[parameters('secretName')]"
+        }
+    }
+}
```
Comment (zhengchang907) [inline arm-oraclelinux-wls-cluster/src/main/arm/nestedtemplates/_keyvault/_keyvaultWithNewCertTemplate.json]: new line

File: arm-oraclelinux-wls-cluster/src/main/arm/nestedtemplates/_keyvaultAdapterTemplate.json [hunk 26]
```diff
@@ -0,0 +1,213 @@
...
+        "sslCertDataSecretName": {
+            "type": "string",
+            "value": "[if(parameters('useExistingAppGatewaySSLCertificate'), reference('keyVaultwithExistingAppGatewaySSLCert', '${azure.apiVersion}').outputs.sslCertDataSecretName.value, reference('keyVaultwithSelfSignedAppGatewaySSLCert', '${azure.apiVersion}').outputs.secretName.value)]"
+        },
+        "sslCertPwdSecretName": {
+            "type": "string",
+            "value": "[if(parameters('useExistingAppGatewaySSLCertificate'), reference('keyVaultwithExistingAppGatewaySSLCert', '${azure.apiVersion}').outputs.sslCertPwdSecretName.value, '')]"
+        }
+    }
+}
```
Comment (zhengchang907) [inline arm-oraclelinux-wls-cluster/src/main/arm/nestedtemplates/_keyvaultAdapterTemplate.json]: new line

File: arm-oraclelinux-wls-cluster/src/main/arm/nestedtemplates/_keyvaultAppGatewayConnectorTemplate.json [hunk 27]
```diff
@@ -0,0 +1,357 @@
...
+        },
+        "adminVMName": {
+            "defaultValue": "adminVM",
+            "type": "string",
+            "metadata": {
+                "description": "Admin Server hosting VM name."
+            }
+        },
+        "appGatewayCertificateOption": {
+            "defaultValue": "",
```
Comment (zhengchang907) [inline arm-oraclelinux-wls-cluster/src/main/arm/nestedtemplates/_keyvaultAppGatewayConnectorTemplate.json]: haveCert

File: arm-oraclelinux-wls-cluster/src/main/arm/nestedtemplates/_keyvaultAppGatewayConnectorTemplate.json [hunk 28]
```diff
@@ -0,0 +1,357 @@
...
+                "haveKeyVault",
+                "haveCert",
+                "generateCert"
+            ]
+        },
+        "domainLabelforApplicationGateway": {
+            "defaultValue": "wlsgw",
+            "type": "string",
+            "metadata": {
+                "description": "DNS for ApplicationGateway"
```
Comment (zhengchang907) [inline arm-oraclelinux-wls-cluster/src/main/arm/nestedtemplates/_keyvaultAppGatewayConnectorTemplate.json]: Application Gateway

File: arm-oraclelinux-wls-cluster/src/main/arm/nestedtemplates/_keyvaultAppGatewayConnectorTemplate.json [hunk 29]
```diff
@@ -0,0 +1,357 @@
...
+            "type": "string",
+            "metadata": {
+                "description": "Public IP Name for the Application Gateway"
+            }
+        },
+        "keyVaultName": {
+            "defaultValue": "",
+            "type": "string",
+            "metadata": {
+                "description": "KeyVault Name"
```
Comment (zhengchang907) [inline arm-oraclelinux-wls-cluster/src/main/arm/nestedtemplates/_keyvaultAppGatewayConnectorTemplate.json]: Key Vault Name

File: arm-oraclelinux-wls-cluster/src/main/arm/nestedtemplates/_keyvaultAppGatewayConnectorTemplate.json [hunk 30]
```diff
@@ -0,0 +1,357 @@
...
+            }
+        },
+        "keyVaultNamefromDeploymentOutput": {
+            "defaultValue": "",
+            "type": "string",
+            "metadata": {
+                "description": "KeyVault Name that output from Key Vault linked template deployment."
```
Comment (zhengchang907) [inline arm-oraclelinux-wls-cluster/src/main/arm/nestedtemplates/_keyvaultAppGatewayConnectorTemplate.json]: Key Vault Name

File: arm-oraclelinux-wls-cluster/src/main/arm/nestedtemplates/_keyvaultAppGatewayConnectorTemplate.json [hunk 31]
```diff
@@ -0,0 +1,357 @@
...
+            }
+        },
+        "keyVaultResourceGroup": {
+            "defaultValue": "",
+            "type": "string",
+            "metadata": {
+                "description": "Resource group name in current subscription containing the KeyVault"
```
Comment (zhengchang907) [inline arm-oraclelinux-wls-cluster/src/main/arm/nestedtemplates/_keyvaultAppGatewayConnectorTemplate.json]: Key Vault

File: arm-oraclelinux-wls-cluster/src/main/arm/nestedtemplates/_keyvaultAppGatewayConnectorTemplate.json [hunk 32]
```diff
@@ -0,0 +1,357 @@
...
+            }
+        },
+        "keyVaultSSLCertDataSecretName": {
+            "defaultValue": "myCertSecretData",
+            "type": "string",
+            "metadata": {
+                "description": "The name of the secret in the specified KeyVault whose value is the SSL Certificate Data,"
```
Comment (zhengchang907) [inline arm-oraclelinux-wls-cluster/src/main/arm/nestedtemplates/_keyvaultAppGatewayConnectorTemplate.json]: Key Vault

File: arm-oraclelinux-wls-cluster/src/main/arm/nestedtemplates/_keyvaultAppGatewayConnectorTemplate.json [hunk 33]
```diff
@@ -0,0 +1,357 @@
...
+            }
+        },
+        "keyVaultSSLCertDataSecretNamefromDeploymentOutput": {
+            "defaultValue": "",
+            "type": "string",
+            "metadata": {
+                "description": "The name of the secret in the specified KeyVault whose value is the SSL Certificate Data, that output from Key Vault linked template deployment."
```
Comment (zhengchang907) [inline arm-oraclelinux-wls-cluster/src/main/arm/nestedtemplates/_keyvaultAppGatewayConnectorTemplate.json]: Key Vault

File: arm-oraclelinux-wls-cluster/src/main/arm/nestedtemplates/_keyvaultAppGatewayConnectorTemplate.json [hunk 34]
```diff
@@ -0,0 +1,357 @@
...
+            }
+        },
+        "keyVaultSSLCertPasswordSecretName": {
+            "defaultValue": "",
+            "type": "string",
+            "metadata": {
+                "description": "The name of the secret in the specified KeyVault whose value is the password for the SSL Certificate"
```
Comment (zhengchang907) [inline arm-oraclelinux-wls-cluster/src/main/arm/nestedtemplates/_keyvaultAppGatewayConnectorTemplate.json]: Key Vault

File: arm-oraclelinux-wls-cluster/src/main/arm/nestedtemplates/_keyvaultAppGatewayConnectorTemplate.json [hunk 35]
```diff
@@ -0,0 +1,357 @@
...
+            }
+        },
+        "keyVaultSSLCertPasswordSecretNamefromDeploymentOutput": {
+            "defaultValue": "",
+            "type": "string",
+            "metadata": {
+                "description": "The name of the secret in the specified KeyVault whose value is the password for the SSL Certificate, that output from Key Vault linked template deployment."
```
Comment (zhengchang907) [inline arm-oraclelinux-wls-cluster/src/main/arm/nestedtemplates/_keyvaultAppGatewayConnectorTemplate.json]: Key Vault

File: arm-oraclelinux-wls-cluster/src/main/arm/nestedtemplates/_keyvaultAppGatewayConnectorTemplate.json [hunk 36]
```diff
@@ -0,0 +1,357 @@
...
+            "maxValue": 5,
+            "metadata": {
+                "description": "Number of VMs to deploy, limit 5 since this sample is using a single storage account"
+            }
+        },
+        "wlsDomainName": {
+            "defaultValue": "wlsd",
+            "type": "string",
+            "metadata": {
+                "description": "Provide Weblogic domain name"
```
Comment (zhengchang907) [inline arm-oraclelinux-wls-cluster/src/main/arm/nestedtemplates/_keyvaultAppGatewayConnectorTemplate.json]: WebLogic

File: arm-oraclelinux-wls-cluster/src/main/arm/nestedtemplates/_keyvaultAppGatewayConnectorTemplate.json [hunk 37]
```diff
@@ -0,0 +1,357 @@
...
+            }
+        },
+        "wlsUserName": {
+            "defaultValue": "weblogic",
+            "type": "string",
+            "metadata": {
+                "description": "Username for your Weblogic domain name"
```
Comment (zhengchang907) [inline arm-oraclelinux-wls-cluster/src/main/arm/nestedtemplates/_keyvaultAppGatewayConnectorTemplate.json]: WebLogic

File: arm-oraclelinux-wls-cluster/src/main/arm/nestedtemplates/_keyvaultAppGatewayConnectorTemplate.json [hunk 38]
```diff
@@ -0,0 +1,357 @@
...
+            }
+        },
+        "wlsPassword": {
+            "type": "securestring",
+            "metadata": {
+                "description": "Password for your Weblogic domain name"
```
Comment (zhengchang907) [inline arm-oraclelinux-wls-cluster/src/main/arm/nestedtemplates/_keyvaultAppGatewayConnectorTemplate.json]: WebLogic

File: arm-oraclelinux-wls-cluster/src/main/arm/nestedtemplates/clusterTemplate.json [hunk 39]
```diff
@@ -459,6 +459,9 @@
                     }
                 }
             },
+            "identity": {
+                "type": "SystemAssigned"
```
Comment (zhengchang907) [inline arm-oraclelinux-wls-cluster/src/main/arm/nestedtemplates/clusterTemplate.json]: remove

File: arm-oraclelinux-wls-cluster/src/main/arm/nestedtemplates/dnszonesTemplate.json [hunk 40]
```diff
@@ -0,0 +1,225 @@
...
+            "type": "array",
+            "metadata": {
+                "description": "An array of CNAME record alias. Used to create CNAME records."
+            },
+            "defaultValue": []
+        },
+        "dnszonesCNAMERecordSetNames": {
+            "type": "array",
+            "metadata": {
+                "description": "An array of A record names. Used to create CNAME record set."
```
Comment (zhengchang907) [inline arm-oraclelinux-wls-cluster/src/main/arm/nestedtemplates/dnszonesTemplate.json]: An array of CNAME record names.

File: arm-oraclelinux-wls-cluster/src/main/arm/nestedtemplates/dnszonesTemplate.json [hunk 41]
```diff
@@ -0,0 +1,225 @@
...
+            "type": "string",
+            "metadata": {
+                "description": "Location for all resources."
+            }
+        },
+        "resourceGroup": {
+            "type": "string",
+            "defaultValue": "[resourceGroup()]",
+            "metadata": {
+                "description": "The resource group name that has an Azure DNS Zone deployed."
```
Comment (zhengchang907) [inline arm-oraclelinux-wls-cluster/src/main/arm/nestedtemplates/dnszonesTemplate.json]: Name of the resource group that has ...

File: arm-oraclelinux-wls-cluster/src/main/arm/nestedtemplates/dnszonesTemplate.json [hunk 42]
```diff
@@ -0,0 +1,225 @@
...
+                    "contentVersion": "1.0.0.0",
+                    "resources": [
+                    ]
+                }
+            }
+        }
+    ],
+    "outputs": {
+    }
+}
```
Comment (zhengchang907) [inline arm-oraclelinux-wls-cluster/src/main/arm/nestedtemplates/dnszonesTemplate.json]: new line

File: arm-oraclelinux-wls-cluster/src/main/scripts/updateDNSZones.sh [hunk 43]
```diff
@@ -0,0 +1,97 @@
...
+                    exit 1
+                fi
+            fi
+
+            count=$((count + 1))
+        done
+
+        index=$((index + 1))
+    done
+fi
```
Comment (zhengchang907) [inline arm-oraclelinux-wls-cluster/src/main/scripts/updateDNSZones.sh]: new line

**Review by zhengchang907 — COMMENTED** [review]
fix

**Comment by galiacheng** [comment]
Close the pr and create a new one for Ed review.

**Pull request closed without merging**